### PR TITLE
ORC-975: Avoid double counting closestFixedBits

### PR DESCRIPTION
**Pull request by guiyanakuang**
### What changes were proposed in this pull request?

```
310  int idx = encodeBitWidth(findClosestNumBits(data[i]));
```

```
public int encodeBitWidth(int n) {
    n = getClosestFixedBits(n);
    .....
}
```

```
public int findClosestNumBits(long value) {
  final int numBits = 64 - Long.numberOfLeadingZeros(value);
  return getClosestFixedBits(numBits);
}
```

getClosestFixedBits is called twice.
Extracts the new method findNumBits from findClosestNumBits. Avoid double counting closestFixedBits.

### Why are the changes needed?

Avoid redundant calls.

### How was this patch tested?

Pass the CIs.

### CODE REVIEW

File: java/core/src/java/org/apache/orc/impl/SerializationUtils.java [hunk 1]
```diff
@@ -270,8 +270,16 @@ public static BigInteger readBigInteger(InputStream input) throws IOException {
...
-    return getClosestFixedBits(numBits);
+    return getClosestFixedBits(findNumBits(value));
+  }
+
+  /**
+   * Count the number of bits left after subtracting leading zeros
+   * @param value
+   * @return bits left after subtracting leading zeros
+   */
+  public int findNumBits(long value) {
```
Comment (dongjoon-hyun) [inline java/core/src/java/org/apache/orc/impl/SerializationUtils.java:281]: Do you have a reference for this naming, @guiyanakuang ? Can we have a more specific name?

Comment (guiyanakuang) [reply java/core/src/java/org/apache/orc/impl/SerializationUtils.java:281]: I've looked up some references and I still don't know what the name of the complement code to remove the high zeros is. How about `validNumBits`.

Comment (dongjoon-hyun) [reply java/core/src/java/org/apache/orc/impl/SerializationUtils.java:281]: Ya, it's tricky. Then, shall we make this `private`? In that case, we don't need to care much about naming.

Comment (guiyanakuang) [reply java/core/src/java/org/apache/orc/impl/SerializationUtils.java]: Yes, I should not expose this internal method.  Fix in 6fa50b2

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @guiyanakuang .
Merged to main/1.7.

**Comment by dongjoon-hyun** [comment]
This is not backported to branch-1.7 because there is a conflict.

**Pull request merged**